core-java-modules/core-java-concurrency-collections-2/src/main/java/com/baeldung/concurrent/lock/SingleLock.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.concurrent.lock;
+
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantLock;
+
+import com.google.common.base.Supplier;
+
+public class SingleLock extends ConcurrentAccessExperiment {
+    ReentrantLock lock;
+
+    public SingleLock() {
+        lock = new ReentrantLock();
+    }
+
+    protected Supplier<?> putSupplier(Map<String,String> map, int key) {
+        return (()-> {
+            try {
+                lock.lock();
```

Comment:
We don't want to lock inside the try (see the [JavaDoc](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/ReentrantLock.html)). The reason is that if lock.lock() did in fact throw a RuntimeException, then the unlock could cause some unexpected results (since it's re-entrant, it could unlock one more level than intended, for example).

Reply:
Ok!